### ISSUE TITLE
[bug] Virtual repository breaks when remote repository becomes unreachable (and conan center is down)

### ISSUE THREAD

**Issue by rconde01** [conan-io/conan#12687]
### Environment Details (include every applicable attribute)
  * Operating System+version: all
  * Compiler+version: n/a
  * Conan version: 1.55
  * Python version: 3.11

### Steps to reproduce (Include if Applicable)
Conan center appears to be down at the moment. I have a virtual repository on Artifactory that uses conan center as a remote repository. I thought some of the point here was to have a cache that was under our control, so if conan center goes down or goes away, we can continue to operate. Shouldn't this work?

**Comment by blackliner** [comment]
wow. This. Is. Bad... @memsharded 

Removing the remote from the virtual repo in Artifactory did not help either, although the package is definitely available in the local repo. This is a death sentence for conan, seriously!
![image](https://user-images.githubusercontent.com/23190927/206924562-4e72ddff-79fd-4d90-931b-93cf67ca4111.png)
![image](https://user-images.githubusercontent.com/23190927/206924564-7d284f81-ddec-4786-9597-562a7739db04.png)

**Comment by memsharded** [comment]
Yes, seems ConanCenter is down, team is investigating.

**Comment by memsharded** [comment]
https://status.cloud.google.com/ it seems GoogleCloud (used by ConanCenter) is having issues, a bit outside of our reach.

**Comment by rconde01** [comment]
That's fair - but to the larger issue, shouldn't this have worked when setup via Artifactory as a remote repository?

**Comment by memsharded** [comment]
> That's fair - but to the larger issue, shouldn't this have worked when setup via Artifactory as a remote repository?

Yes, we need to follow up with the Artifactory team to better understand this issue. Thanks very much for reporting it.

**Comment by memsharded** [comment]
The incident with ConanCenter seems to be resolved. Please try again and let us know.

**Comment by kymikoloco** [comment]
https://www.jfrog.com/confluence/display/JFROG/Remote+Repositories#RemoteRepositories-SingleRepositoryOffline

Manually setting the remote to **Offline** may be a requirement to bypass the query and allow the caching to work in these situations, albeit temporarily. Not ideal, I would also have assumed it would use the cache as a fallback, especially if there are local packages in the virtual repo unrelated to the remote, but I'm not intimately familiar with the backend of Artifactory in these situations.

**Comment by rconde01** [comment]
any feedback here from the artifactory team?

**Comment by blackliner** [comment]
They reached out to me through their commercial support channel, but other than a few misplaced explanations, no solution.

**Comment by memsharded** [comment]
I am discussing the topic with the Artifactory teams.
At the moment it seems that it is in fact necessary to go to Artifactory and activate "offline mode", but there is no such automatic activation of "offline mode" when the proxied repo (ConanCenter in this case) is down.

I will keep pushing this internally, but at the moment the alternatives could be:
- Live with the potential occasional disruptions of ConanCenter, they are unlikely but they might happen from time to time, for example sometimes we are being affected by other cloud providers outages or issues. Have some admin that can quickly activate "offline mode" when necessary.
- Decouple the ConanCenter remote via the Conan client: define 2 repos in the client and do ``conan download -r=conancenter`` + ``conan upload -r=myrepo`` of the desired artifacts when desired, populate your own ``myrepo`` remote that is the one used in production. This approach needs onboarding new packages that might be necessary explicitly, and also explicitly doing updates.
- Decouple the ConanCenter remote by building your own binaries from ``conan-center-index`` repo and uploading to your own remote. This approach is the recommended one for most enterprises, as it allows full control of what goes into the company binaries, full ownership of binaries, full isolation of automatic updates in the upstream with much better robustness and security. This can be achieved quite easily with some simple scripts like the ones I have in my fork: https://github.com/memsharded/conan-center-index/tree/mycustom (``mybuild.py``).

**Comment by memsharded** [comment]
Team is trying to reproduce this issue, but no success. It seems that latest Artifactory version is doing the expected and is able to serve cached packages even hen the proxied repository (ConanCenter is down). 
Can you please report which Artifactory versions are you running? Thanks.

**Comment by blackliner** [comment]
How did you reproduce the issue?

**Comment by memsharded** [comment]
Fetching packages from a remote repo then disconnecting the network to the proxied repo, re-installing from that remote repo.

**Comment by rconde01** [comment]
I’m not in a position test at the moment…but I wonder what happens if it’s reachable but  unresponsive. In my case it shouldn’t have needed anything from the actual conan remote.

**Comment by memsharded** [comment]
> I’m not in a position test at the moment…but I wonder what happens if it’s reachable but unresponsive. In my case it shouldn’t have needed anything from the actual conan remote.

Yes, this is what I am suspecting, that the behavior might not be exactly the same if it is a "clean" disconnection or if there are other kind of issues. Specially if it not responsive, that might be definitely a very complicated issue to resolve.

**Comment by memsharded** [comment]
Closing this as outdated, and in any case, it is better to push this kind of issue directly against Artifactory, as this is something that fully belongs there, there is nothing that can be done from here, the Conan client side. Many thanks for the feedback.

Also, the model has pretty shifted since then, using a Remote repo to proxy ConanCenter is not necessarily the most recommended approach, but instead directly building binaries from ``conan-center-index`` repo, see https://docs.conan.io/2/devops/using_conancenter.html
